src/common/charset/Charset.cpp

```diff
@@ -0,0 +1,178 @@
+                                           const std::string& collateName) {
+    auto iter = charsetToCollation.find(charsetName);
+    if (iter != charsetToCollation.end()) {
+        for (auto& sc : iter->second.supportColl_) {
+            if (!sc.compare(collateName)) {
+                return Status::OK();
+            }
+        }
+    }
+    return  Status::Error("Charset `%s' and Collation `%s' not match",
+                          charsetName.c_str(), collateName.c_str());
+}
+
+
+StatusOr<std::string> CharsetInfo::getDefaultCollationbyCharset(const std::string& charsetName) {
+    auto iter = charsetToCollation.find(charsetName);
+    if (iter != charsetToCollation.end()) {
+        return iter->second.defaultColl_;
+    }
+    return Status::Error("Charset `%s' not support", charsetName.c_str());
+}
+
+
+StatusOr<std::string> CharsetInfo::getCharsetbyCollation(const std::string& collationName ) {
+    for (auto& cset : charsetToCollation) {
+        for (auto& coll : cset.second.supportColl_) {
+            if (!coll.compare(collationName)) {
+                return cset.first;
+            }
+        }
+    }
+    return Status::Error("Collation `%s' not support", collationName.c_str());
+}
+
+
+StatusOr<int>
+CharsetInfo::nebulaStrCmp(const std::string& collateName, const char* p1, const char* p2) {
+    auto iter = CharsetInfo::collateToLocale.find(collateName);
+    if (iter == CharsetInfo::collateToLocale.end()) {
+        return Status::Error("Collation `%s' not support", collateName.c_str());
+    }
+
+    // For unset LC_ALL, May occur "locale::facet::_S_create_c_locale name not valid"
+    std::string curLoc = setlocale(LC_ALL, NULL);
+    if (curLoc.empty()) {
+        setlocale(LC_ALL, "en_US.UTF-8");
+    }
+
+    std::locale loc(iter->second);
+    if (loc.name().empty()) {
+        return Status::Error("The locale variable is empty");
+    }
+    auto& f = std::use_facet<std::collate<char>>(loc);
+
+    std::string str1(p1), str2(p2);
+    return f.compare(&str1[0], &str1[0] + str1.size(),
+                     &str2[0], &str2[0] + str2.size());
+}
+
+
+StatusOr<bool>
+CharsetInfo::nebulaStrCmpLT(const std::string& collateName, const char* p1, const char* p2) {
+    auto ret = nebulaStrCmp(collateName, p1, p2);
+    if (!ret.ok()) {
+        return ret.status();
+    }
+    auto val = ret.value();
+    if (val < 0) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+
+StatusOr<bool>
+CharsetInfo:: nebulaStrCmpLE(const std::string& collateName, const char* p1, const char* p2) {
+    auto ret = nebulaStrCmp(collateName, p1, p2);
+    if (!ret.ok()) {
+        return ret.status();
+    }
+    auto val = ret.value();
+    if (val <= 0) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+
+StatusOr<bool>
+CharsetInfo::nebulaStrCmpGT(const std::string& collateName, const char* p1, const char* p2) {
+    auto ret = nebulaStrCmpLE(collateName, p1, p2);
```

Comment:
what if just return  nebulaStrCmpLE(collateName, p2, p1) ?

Reply:
Yeah,  this is equivalent